main.go

```diff
@@ -72,7 +83,7 @@ func main() {
 		dialer = dialerPerHost
 	}
 
-	proxyServer := proxy.NewServer(dialer)
+	proxyServer := proxy.NewServer(dialer, sm.Hash)
```

Comment:
api.Server has Run method but proxy.Server is used as a handler and passed to http.ListenAndServe directly. Maybe they should both look the same?

Reply:
Both are API and Proxy server are calling `ListenAndServe` from `http.Server`.